docs/custom_promise_types/modules.md

```diff
@@ -120,10 +120,21 @@ In short, when writing a promise module, these log levels should be used:
 
 * `critical` - Serious errors in protocol or module itself (not in policy)
 * `error` - Errors when validating / evaluating a promise, including syntax errors and promise not kept
-* `info` - Changes made (repaired promises)
+* `warning` - The promise did not fail, but there is something the user (policy writer) should probably fix. Some examples:
+  * Policy relies on deprecated behavior/syntax which will change
+  * Policy uses demo / unsafe options which should be avoided in a production environment
+* `notice` - Unusual events which you want to notify the user about
+  * Most promise types won't need this - usually `info` or `warning` is more appropriate
+  * Useful for events which happen rarely and are not the result of a promise, for example:
+    * New credentials detected
+    * New host bootstrapped
+    * The module made a change to the system for itself to work (database initialized, user created)
+* `info` - Changes made to the system (1 per repaired promise)
```

Comment:
Why *1 per repaired promise*?

Reply:
We could add a "usually" or something. It's a good rule to follow in general. 0 is too few 2 is too many :)

Reply:
I can imagine the promises handled by promise modules more than any other promises will be making multiple changes at once.

Reply:
Added a clarifying sentence: https://github.com/cfengine/core/pull/4476/commits/13f4d84033fad96f2d23e9bbfa6485a4623bd61e